docs/datamodel/ols.rst

```diff
@@ -0,0 +1,261 @@
+  ...  };
+  {default::BlogPost {id: e76afeae-03db-11ed-b346-fbb81f537ca6}}
+
+We've created a ``User``, set the value of ``current_user`` to its ``id``, and
+created a new ``BlogPost``. When we try to select all ``BlogPost`` objects,
+we'll see the post we just created.
+
+.. code-block:: edgeql-repl
+
+  db> select BlogPost;
+  {default::BlogPost {id: e76afeae-03db-11ed-b346-fbb81f537ca6}}
+  db> select count(BlogPost);
+  {1}
+
+Now let's unset ``current_user`` and see what happens.
+
+.. code-block:: edgeql-repl
+
+  db> set global current_user := {};
+  OK: SET GLOBAL
+  db> select BlogPost;
+  {}
+  db> select count(BlogPost);
+  {0}
+
+Now ``select BlogPost`` returns zero results. We can only ``select`` the
+*posts* written by the *user* specified by ``current_user``. When
+``current_user`` has no value, we can't read any posts.
+
+The access policies use global variables to define a "subgraph" of data that
+is visible to a particular query.
+
+Policy types
+^^^^^^^^^^^^
+
+For the most part, the policy types correspond to EdgeQL's *statement types*:
+
+- ``select``: Applies to all queries; objects without a ``select`` permission
+  cannot be modified either.
+- ``insert``: Applies to insert queries; executed *post-insert*. If an
+  inserted object violates the policy, the query will fail.
+- ``delete``: Applies to delete queries.
+- ``update``: Applies to update queries.
+
+Additionally, the ``update`` operation can broken down into two sub-policies:
+``update read`` and ``update write``.
+
+- ``update read``: This policy restricts *which* objects can be updated. It
+  runs *pre-update*; that is, this policy is executed before the updates have
+  been applied.
+- ``update write``: This policy restricts *how* you update the objects; you
+  can think of it as a *post-update* validity check. This could be used to
+  prevent a ``User`` from transferring a ``BlogPost`` to another ``User``.
+
+Finally, there's an umbrella policy that can be used as a shorthand for all
+the others.
+
+- ``all``: A shorthand policy that can be used to allow or deny full read/
+  write permissions. Exactly equivalent to ``select, insert, update, delete``.
+
+
+Resolution algorithm
+^^^^^^^^^^^^^^^^^^^^
+
+An object type can contain an arbitrary number of access policies, including
+several conflicting ``allow`` and ``deny`` policies. EdgeDB uses a particular
+algorithm for resolving these policies.
+
+.. figure:: images/ols.png
+
+  The access policy resolution algorithm, explained with Venn diagrams.
+
+1. When no policies are defined on a given object type,
+   all objects of that type can be read or modified by any appropriately
+   authenticated connection.
+
+2. EdgeDB then applies all ``allow`` policies. Each policy grants a
+   *permission* that is scoped to a particular *set of objects*. Conceptually,
+   these permissions are merged with the ``union`` / ``or`` operator to
+   determine the set of allowable actions.
```

Comment:
1.1. If any access policy is defined on an object type *no objects* of that type can now be accessed in any way unless a policy explicitly allows it.